src/tangle.ts

```diff
@@ -136,20 +137,13 @@ export class Client<T> {
             fn(this.defaultValue[eventName]);
         }
 
-        let lastValue: T[K];
         return this.events.pipe(
             map(event => event?.transient),
             map(transient => transient?.[eventName] as T[K]),
             filter(value => typeof value !== 'undefined'),
-            map(value => value as T[K])
-        ).subscribe((val: T[K]) => {
-            if (val === lastValue) {
-                return;
-            }
-
-            lastValue = val;
-            fn(val);
-        });
+            map(value => value as T[K]),
+            distinctUntilChanged((prev, curr) => prev === curr)
```

Comment:
you can just do `distinctUntilChanged()` since the default comparator will do the job.

Reply:
Btw, if we want complex types (ref vs val) other than numbers and strings to work we’d need a JSON stringify comparator which could get expensive unless we really need it.